### PR TITLE
Fixes a restoration edge-case.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Fixes [this issue](https://github.com/wordpress-mobile/WordPress-iOS-Editor/issues/560).

It seems there's some strange problem with the legacy restoration code, when the new editor is enabled.  In order to fix this issue in a secure manner I disabled the legacy editor restoration if the new editor should be used instead - the result is that the post will be saved as draft but the legacy editor window will not be restored.

/cc @SergioEstevao

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 1]
```diff
@@ -340,27 +340,55 @@ + (UIViewController *)viewControllerWithRestorationIdentifierPath:(NSArray *)ide
...
+{
+    UINavigationController *navController = [[UINavigationController alloc] init];
+    navController.restorationIdentifier = WPEditorNavigationRestorationID;
+    navController.restorationClass = self;
+    
+    return navController;
+}
+
++ (UIViewController*)restoreViewControllerWithIdentifierPath:(NSArray*)identifierComponents
+                                                       coder:(NSCoder *)coder
```
Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: Style brigade: the start needs to cuddle the variable type

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 2]
```diff
@@ -340,27 +340,55 @@ + (UIViewController *)viewControllerWithRestorationIdentifierPath:(NSArray *)ide
...
-        
+        restoredViewController = [self restoreParentNavigationController];
     } else if ([self isSelfIdentifierPath:identifierComponents]) {
+        restoredViewController = [self restoreViewControllerWithCoder:coder];
+    }
+    
+    return restoredViewController;
+}
+
++ (UIViewController*)restoreViewControllerWithCoder:(NSCoder *)coder
```
Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: style brigade: start cuddle

File: WordPress/Classes/ViewRelated/Post/WPPostViewController.m [hunk 3]
```diff
@@ -340,27 +340,55 @@ + (UIViewController *)viewControllerWithRestorationIdentifierPath:(NSArray *)ide
...
++ (UIViewController*)restoreParentNavigationController
+{
+    UINavigationController *navController = [[UINavigationController alloc] init];
+    navController.restorationIdentifier = WPEditorNavigationRestorationID;
+    navController.restorationClass = self;
+    
+    return navController;
+}
+
++ (UIViewController*)restoreViewControllerWithIdentifierPath:(NSArray*)identifierComponents
```
Comment (SergioEstevao) [inline WordPress/Classes/ViewRelated/Post/WPPostViewController.m]: (NSArray *)identifierComponents

**Comment by diegoreymendez** [comment]
@SergioEstevao - As discussed over slack, I just changed the parameter that was not following our style guide.

**Comment by SergioEstevao** [comment]
Just tested it and it's ready to go :shipit:

**Pull request merged**